xbmc/network/WakeOnAccess.cpp

```diff
@@ -70,6 +75,118 @@ static unsigned long HostToIP(const std::string& host)
   return inet_addr(ip.c_str());
 }
 
+#define LOCALIZED(id) g_localizeStrings.Get(id)
+
+static void ShowDiscoveryMessage(const char* function, const char* server_name, bool new_entry)
+{
+  std::string message;
+
+  if (new_entry)
+  {
+    CLog::Log(LOGINFO, "%s - Create new entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13035).c_str(), server_name);
+  }
+  else
+  {
+    CLog::Log(LOGINFO, "%s - Update existing entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13034).c_str(), server_name);
+  }
+  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, LOCALIZED(13033), message, 4000, true, 3000);
+}
+
+struct UPnPServer
+{
+  UPnPServer()
+  {
+    m_nextWake = CDateTime::GetCurrentDateTime();
+  }
+  bool operator == (const UPnPServer& server) const { return server.m_uuid == m_uuid; }
+  bool operator != (const UPnPServer& server) const { return !(*this == server); }
+  bool operator == (const std::string& server_uuid) const { return server_uuid == m_uuid; }
+  bool operator != (const std::string& server_uuid) const { return !(*this == server_uuid); }
+  std::string m_name;
+  std::string m_uuid;
+  std::string m_mac;
+  CDateTime m_nextWake;
+};
+
+typedef std::vector<UPnPServer> upnp_server_list_t;
```

Comment:
This naming doesn't really follow our naming convention. And with C++11 `typedef` is old school ;-)

Reply:
OK I will give it a last shot (I have pretty much lost interest by now)